### PR TITLE
Add NetworkSets support

### PR DESCRIPTION
**Pull request by lmm**
## Description
Add support for NetworkSets.

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
Add NetworkSets
```

### CODE REVIEW

**Comment by lmm** [comment]
Tests to come

File: calicoctl/resourcemgr/networkset.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	client "github.com/projectcalico/libcalico-go/lib/clientv3"
+	"github.com/projectcalico/libcalico-go/lib/options"
+)
+
+func init() {
+	registerResource(
+		api.NewNetworkSet(),
+		api.NewNetworkSetList(),
+		true,
+		[]string{"networkset", "networksets", "ns"},
```
Comment (caseydavenport) [inline calicoctl/resourcemgr/networkset.go]: Hrm, the abbreviation "ns" seems nice but is also what Kubernetes uses for "Namespace" so might be confusing... especially if we decide to add a Namespace object some day.

Maybe `netset`? WDYT?

**Comment by caseydavenport** [comment]
This looks like it's probably right to me- once we get the libcalico PR merged we can bump the version in here to refer to master and hopefully it all just works!

File: calicoctl/resourcemgr/networkset.go (hunk 1 above)
Comment (lmm) [reply calicoctl/resourcemgr/networkset.go]: Ahh that's a good point. `netset` is nice. And for GlobalNetworkSets, `gnetset`?

**Pull request merged**